server/internal/usecase/interfaces/model.go

```diff
@@ -0,0 +1,38 @@
+package interfaces
+
+import (
+	"context"
+	"errors"
+
+	"github.com/reearth/reearth-cms/server/internal/usecase"
+	"github.com/reearth/reearth-cms/server/pkg/id"
+	"github.com/reearth/reearth-cms/server/pkg/model"
+)
+
+type CreateModelParam struct {
+	ProjectId   id.ProjectID
+	Name        *string
+	Description *string
+	Key         *string
+}
+
+type UpdateModelParam struct {
+	ModelId     id.ModelID
+	Name        *string
+	Description *string
+	Key         *string
+}
+
+var (
+	ErrModelKey error = errors.New("model key is already used by another model")
+)
+
+type Model interface {
+	Fetch(context.Context, []id.ModelID, *usecase.Operator) (model.List, error)
```

Comment:
```suggestion
	FindByIDs(context.Context, []id.ModelID, *usecase.Operator) (model.List, error)
```

I think FindBy~ is better